sdk/internal/testframework/recording_sanitizer.go

```diff
@@ -0,0 +1,85 @@
+// +build go1.13
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package testframework
+
+import (
+	"net/http"
+
+	"github.com/dnaeon/go-vcr/cassette"
+	"github.com/dnaeon/go-vcr/recorder"
+)
+
+type RecordingSanitizer struct {
+	recorder          *recorder.Recorder
+	headersToSanitize map[string]*string
+	urlSanitizer      StringSanitizer
+	bodySanitizer     StringSanitizer
+}
+
+type StringSanitizer func(*string)
+
+const SanitizedValue string = "sanitized"
+const SanitizedBase64Value string = "Kg=="
+
+var sanitizedValueSlice = []string{SanitizedValue}
+
+// defaultSanitizer returns a new RecordingSanitizer with the default sanitizing behavior.
+// To customize sanitization, call AddSanitizedHeaders, AddBodySanitizer, or AddUrlSanitizer.
+func defaultSanitizer(recorder *recorder.Recorder) *RecordingSanitizer {
+	// The default sanitizer sanitizes the Authorization header
+	s := &RecordingSanitizer{headersToSanitize: map[string]*string{"Authorization": nil}, recorder: recorder, urlSanitizer: DefaultStringSanitizer, bodySanitizer: DefaultStringSanitizer}
+	recorder.AddSaveFilter(s.applySaveFilter)
+
+	return s
+}
+
+// AddSanitizedHeaders adds the supplied header names to the list of headers to be sanitized on request and response recordings.
+func (s *RecordingSanitizer) AddSanitizedHeaders(headers ...string) {
+	for _, headerName := range headers {
+		s.headersToSanitize[headerName] = nil
+	}
+}
+
+// AddBodysanitizer configures the supplied StringSanitizer to sanitize recording request and response bodies
+func (s *RecordingSanitizer) AddBodysanitizer(sanitizer StringSanitizer) {
+	s.bodySanitizer = sanitizer
+}
+
+// AddUriSanitizer configures the supplied StringSanitizer to sanitize recording request and response URLs
+func (s *RecordingSanitizer) AddUrlSanitizer(sanitizer StringSanitizer) {
+	s.urlSanitizer = sanitizer
+}
+
+func (s *RecordingSanitizer) sanitizeHeaders(header http.Header) {
+	for headerName := range s.headersToSanitize {
```

Comment:
While not necessarily important for testing, iterating over a map is fairly expensive in Go.

Reply:
That's surprising - is there a more efficient way to do this?

Reply:
If you're not using the map as a hash set, it might be better to just use a slice of headers to sanitize instead.  For this case I don't think it really matters, I just wanted to point out that this can be problematic in some cases (I'm acutely aware of this from profiling msal/go where this happens in a few places and it shows up fairly dominant).

Reply:
I guess we aren't using it as a hash in this case, since we hash lookup in the request, but only iterate through the sanitized header names. Good callout!